common/src/test/java/com/itsaky/androidide/utils/ZipUtilsTest.kt

```
@@ -58,4 +60,34 @@ class ZipUtilsTest {
 	}
+
+	@Test
+	fun `unzipFile refuses to extract over an existing symlink`() {
+		val destDir = tempFolder.newFolder("dest")
+		val realFile = File(destDir, "real.txt").apply { writeText("original") }
+		val linkPath = File(destDir, "link.txt").toPath()
+		val symlinkCreated =
+			try {
+				Files.createSymbolicLink(linkPath, realFile.toPath())
+				true
+			} catch (e: UnsupportedOperationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)